From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Fri, 19 Jan 2018 16:31:01 +0100
Subject: workspaceThumbnail: only update _porthole if the overview is visible

Otherwise it happens that porthole is computed again after that the
overlay is hidden (triggered by a layout reallocation) and thus not
regenerated again afterwards.

(cherry picked from commit 5fcf40b973286fd442148ee3348becd493c8206d)
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=792687
Origin: upstream, 3.28.2, commit:b476e851b709445f8c0fb0eb7ce34454318cad6d
---
 js/ui/workspaceThumbnail.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/ui/workspaceThumbnail.js b/js/ui/workspaceThumbnail.js
index 78fac80..c1b4bdd 100644
--- a/js/ui/workspaceThumbnail.js
+++ b/js/ui/workspaceThumbnail.js
@@ -1168,7 +1168,7 @@ var ThumbnailsBox = new Lang.Class({
     // The "porthole" is the portion of the screen that we show in the
     // workspaces
     _ensurePorthole() {
-        if (!Main.layoutManager.primaryMonitor)
+        if (!Main.layoutManager.primaryMonitor || !Main.overview.visible)
             return false;
 
         if (!this._porthole)
